pkg/manager/manager_github_cri.go

```diff
@@ -0,0 +1,271 @@
+	// }
+	// // Add the JIT config token as an environment variable
+	// cmd.Args = append(cmd.Args, "--env", fmt.Sprintf("WARPBUILD_GH_JIT_TOKEN=%s", opts.JitToken))
+	// // Add the image and command
+	// cmd.Args = append(cmd.Args, m.ContainerOptions.Image)
+	// // Add the command and its arguments
+	// if m.ContainerOptions.Cmd != "" {
+	// 	cmd.Args = append(cmd.Args, m.ContainerOptions.Cmd)
+	// }
+	// cmd.Args = append(cmd.Args, m.ContainerOptions.Args...)
+
+	// cmd.Dir = m.RunnerDir
+
+	// Base command
+	cmd := exec.CommandContext(ctx, m.CMDOptions.CMD, m.CMDOptions.Args...)
+	cmd.Env = append(cmd.Env, "WARPBUILD_GH_JIT_TOKEN="+opts.JitToken)
+	// log.Logger().Infof("Found envs: %v", m.CMDOptions.Envs)
+	for _, env := range m.CMDOptions.Envs {
+		log.Logger().Infof("setting env %s=%s", env.Key, env.Value)
+		cmd.Env = append(cmd.Env, fmt.Sprintf("%s=%s", env.Key, env.Value))
+	}
+	cmd.Dir = m.CMDOptions.Dir
+
+	log.Logger().Infof("starting runner with command: %s", cmd.String())
+	log.Logger().Infof("JIT Token: %s", opts.JitToken)
+
+	stdoutPipe, err := cmd.StdoutPipe()
+	if err != nil {
+		log.Logger().Errorf("error creating stdout pipe: %v", err)
+		return nil, err
+	}
+
+	stderrPipe, err := cmd.StderrPipe()
+	if err != nil {
+		log.Logger().Errorf("error creating stderr pipe: %v", err)
+		return nil, err
+	}
+
+	for _, hook := range GetHooks[IPreStartHook]() {
+		err := hook.PreStartHook(ctx, &PreStartHookOptions{
+			StartRunnerOptions: opts,
+			ManagerOptions: &ManagerOptions{
+				Provider:  ProviderGithubCRI,
+				GithubCRI: m.GithubCRIOptions,
+			},
+		})
+		if err != nil {
+			log.Logger().Errorf("error running pre-start hook %s: %v", hook.HookID(), err)
+			return nil, err
+		}
+	}
+
+	if err := cmd.Start(); err != nil {
+		log.Logger().Errorf("error starting command: %v", err)
+		return nil, err
+	}
+
+	stdoutChan := make(chan string)
+	stderrChan := make(chan string)
+	doneChan := make(chan bool)
+
+	go captureOutput(stdoutPipe, stdoutChan)
+	go captureOutput(stderrPipe, stderrChan)
+
+	// Open files to write stdout and stderr
+	stdoutFile, err := openFile(m.StdoutFile)
+	if err != nil {
+		log.Logger().Errorf("error opening stdout file: %v", err)
+		return nil, err
+	}
+	defer stdoutFile.Close()
+
+	stderrFile, err := openFile(m.StderrFile)
+	if err != nil {
+		log.Logger().Errorf("error opening stderr file: %v", err)
+		return nil, err
+	}
+	defer stderrFile.Close()
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+
+	go func() {
+		defer wg.Done()
+		cmd.Wait()
+		doneChan <- true
+	}()
+
+	// Ticker to capture output every second
+	ticker := time.NewTicker(time.Second)
+	for {
+		select {
+		case out := <-stdoutChan:
+			fmt.Fprintln(stdoutFile, out)
+			fmt.Fprintln(os.Stdout, out)
+		case err := <-stderrChan:
+			fmt.Fprintln(stderrFile, err)
+			fmt.Fprintln(os.Stderr, err)
+		case <-ticker.C:
+			// Handle output every second
```

Comment:
Nothing is happening on this ticker. Should we remove this?

Reply:
removed